tests/SecretsCrypto/tst_secretscrypto/tst_secretscrypto.c

```diff
@@ -1567,6 +1569,449 @@ static void tst_crypto_import_invalid(SfCryptoFixture *fixture, gconstpointer te
+            key,
+            SF_CRYPTO_OPERATION_DECRYPT,
+            kd->block_mode,
+            kd->padding,
+            SF_CRYPTO_SIGNATURE_PADDING_NONE,
+            SF_CRYPTO_DIGEST_SHA256,
+            NULL,
+            CRYPTO_PLUGIN_TEST,
+            NULL,
+            _tst_crypto_ref_res_and_quit,
+            fixture);
+    g_main_loop_run(fixture->loop);
+
+    decoder = sf_crypto_cipher_session_new_finish(fixture->test_res, &fixture->error);
+
+    g_assert_no_error(fixture->error);
+    g_assert_nonnull(decoder);
+
+    if (g_test_failed())
+        return;
+
+    g_test_queue_unref(decoder);
+
+    encoded = g_byte_array_sized_new(kd->data_size * kd->chunks);
+    decoded = g_byte_array_sized_new(kd->data_size * kd->chunks);
+
+    g_test_queue_destroy((GDestroyNotify)g_byte_array_unref, encoded);
+    g_test_queue_destroy((GDestroyNotify)g_byte_array_unref, decoded);
+
+    for (i = 0; i < kd->chunks; i++) {
+        GBytes *random_bytes = NULL;
+        GBytes *encoded_bytes;
+
+        sf_crypto_manager_generate_random_data(fixture->manager,
+                kd->data_size,
+                SF_CRYPTO_DEFAULT_CSPRNG_ENGINE,
+                NULL,
+                CRYPTO_PLUGIN_TEST,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_main_loop_run(fixture->loop);
+
+        random_bytes = sf_crypto_manager_generate_random_data_finish(fixture->test_res, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+        g_assert_nonnull(random_bytes);
+
+        if (g_test_failed())
+            return;
+
+        g_byte_array_append(encoded, g_bytes_get_data(random_bytes, NULL), g_bytes_get_size(random_bytes));
+
+        sf_crypto_cipher_session_update(encoder,
+                random_bytes,
+                NULL,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_bytes_unref(random_bytes);
+        g_main_loop_run(fixture->loop);
+
+        sf_crypto_cipher_session_update_finish(fixture->test_res, &encoded_bytes, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+        g_assert_nonnull(encoded_bytes);
+
+        if (g_test_failed())
+            return;
+
+        sf_crypto_cipher_session_update(decoder,
+                encoded_bytes,
+                NULL,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_bytes_unref(encoded_bytes);
+        g_main_loop_run(fixture->loop);
+
+        sf_crypto_cipher_session_update_finish(fixture->test_res, &random_bytes, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+        g_assert_nonnull(random_bytes);
+
+        if (g_test_failed())
+            return;
+
+        g_byte_array_append(decoded, g_bytes_get_data(random_bytes, NULL), g_bytes_get_size(random_bytes));
+        g_bytes_unref(random_bytes);
+
+        g_assert_cmpmem(encoded->data, MIN(encoded->len, decoded->len),
```

Comment:
I'm not sure I understand what this comparison is doing.  I had thought that encoded->data should point to encrypted data, and decoded->data should point to decrypted data (i.e. identical to the sum of the random_bytes chunks), but if that were the case it looks like this comparison would ensure that the encrypted data is equal to the decrypted data, so I must be confused?

Reply:
Oh, never mind.  encoded->data is the sum of random_bytes chunks, not the encrypted data.  decoded->data is indeed the decrypted data, and thus they should indeed be equal.

Reply:
This is a pretty useless comparison, the full comparison is done at the end of the function. At least with this test setup, the "decoded" array lags behind by some (IIRC 22) bytes. This just confirms that all data received thus far is valid.